PureBasicIDE/FileSystem.pb

```diff
@@ -275,3 +275,23 @@ Procedure.s ResolveRelativePath(BasePath$, FileName$)
   ProcedureReturn UniqueFilename(FileName$)
 EndProcedure
 
+Procedure CreateDirectoryRecursive(Directory$)
+  If Directory$
+    Select FileSize(Directory$)
+      Case -2 ; already exists - OK!
+        ProcedureReturn #True
+        
+      Case -1 ; does not exist - create it
+        CreateDirectoryRecursive(ResolveRelativePath(Directory$ + #Separator + "..", ""))
```

Comment:
This use of `ResolveRelativePath()` does not fit how the procedure is documented above: The first argument is supposed to be a full path (no "..") and only the second argument is supposed to be relative. 

So `ResolveRelativePath(Directory$, ".." + #Separator)` looks more right. 

Then you also need to change the abort condition for the recursion, as the procedure returns the second argument on failure.